roles/cloud-cloudstack/tasks/main.yml

```diff
@@ -0,0 +1,68 @@
+---
+- block:
+    - name: Build python virtual environment
+      import_tasks: venv.yml
+
+    - block:
+      - name: Include prompts
+        import_tasks: prompts.yml
+        
+      - set_fact:
+          algo_region: >-
+            {% if region is defined %}{{ region }}
+            {%- elif _algo_region.user_input is defined and _algo_region.user_input != "" %}{{ cs_zones[_algo_region.user_input | int -1 ]['name'] }}
+            {%- else %}{{ cs_zones[default_region | int - 1]['name'] }}{% endif %}
+
+      - name: Security group created
+        cs_securitygroup:
+          name: "{{ algo_server_name }}-security_group"
+          description: AlgoVPN security group
+        register: cs_security_group
+
+      - name: Security rules created
+        cs_securitygroup_rule:
+          security_group: "{{ cs_security_group.name }}"
+          protocol: "{{ item.proto }}"
+          start_port: "{{ item.start_port }}"
+          end_port: "{{ item.end_port }}"
+          cidr: "{{ item.range }}"
+        with_items:
+          - { proto: tcp, start_port: 22, end_port: 22, range: 0.0.0.0/0 }
+          - { proto: udp, start_port: 4500, end_port: 4500, range: 0.0.0.0/0 }
+          - { proto: udp, start_port: 500, end_port: 500, range: 0.0.0.0/0 }
+          - { proto: udp, start_port: "{{ wireguard_port }}", end_port: "{{ wireguard_port }}", range: 0.0.0.0/0 }
+
+      - name: Keypair created
+        cs_sshkeypair:
+          name: "{{ SSH_keys.comment|regex_replace('@', '_') }}"
+          public_key: "{{ lookup('file', '{{ SSH_keys.public }}') }}"
+        register: cs_keypair
+
+      - name: Set facts
+        set_fact:
+          image_id: "{{ cloud_providers.cloudstack.image }}"
+          size: "{{ cloud_providers.cloudstack.size }}"
```

Comment:
I just found `size` is defined, but not used. I guess it should be defined in `service_offering` for the `cs_instance` module.